### PR TITLE
fix: satisfies expressions are parsed as TsSatisfiesExpression

### LINKED ISSUE #6238: Seems all satisfies expressions are parsed as "as expression"
### Describe the bug

It seems all satisfies expressions are parsed as `TsAsExpression`. The expected test output in the PR for it seem to indicate that as well.

Side note: can we rename `TsSatisfactionExpression` to `TsSatisfiesExpression`? TypeScript calls it `SatisfiesExpression`

### Input code

```typescript
const test = testingtesting satisfies other;
```

### Config

_No response_

### Playground link

https://play.swc.rs/?version=1.3.10&code=H4sIAAAAAAAAA0vOzysuUShJBRK2YCozLx1KKRQnlmQWp2WmFivkl2SkFlkDAN47ODUsAAAA&config=H4sIAAAAAAAAAy2NsQrDMAxE%2F0VzhrZDoV5bsvUjRCIXFcc2OhsaQv69dsh2nN47bfTFRG6jzAaxnrDGwj9yVNYsmExzoYEKWlWsyt4y20dKIwS3y%2FXRr8YRPtnSByrkJV6jjMmegYFRJcw49eYvaa5BOtpfHDt36r1G9Ss5zwEykOJ9gof4B7AXfpKrAAAA

### Expected behavior

Parsed as `TsSatisfactionExpression`.

### Actual behavior

Parsed as `TsAsExpression`.

### Version

1.3.10

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by nissy-dev**
**Description:** Fix #6238 

**BREAKING CHANGE:**

**Related issue (if exists):** #6238

### CODE REVIEW

**Comment by nissy-dev** [comment]
node bindings build failed... Based on the error messages, I seem the old version swc_ecma_parser is compiled when running `yarn build:dev`. I don't understand why this happened. I will look into it.

**Comment by kdy1** [comment]
I think this is a problematic, but you need to patch local crates using 

https://github.com/swc-project/swc/blob/8123bf09b58785f1e972b1e31b312a00d5535f4e/.github/workflows/CI.yml#L716-L719

File: .github/workflows/CI.yml [hunk 1]
```diff
@@ -717,6 +717,7 @@ jobs:
         run: |
           echo '[patch.crates-io]' >> bindings/Cargo.toml
           ./scripts/cargo/patch-section.sh >> bindings/Cargo.toml
+          cd bindings && cargo update
```
Comment (nissy-dev) [inline .github/workflows/CI.yml:720]: This line was needed to apply patch dependencies correctly.

Rust warned:
```
If the patch has a different version from what is locked in the Cargo.lock file, 
run `cargo update` to use the new version.
```

In fact, cargo used the version in Cargo.lock file when building for node-binding test.

File: node-swc/__tests__/transform/issue_4730_test.mjs [hunk 2]
```diff
@@ -37,9 +37,9 @@ Object.defineProperty(exports, \\"__esModule\\", {
     value: true
 });
 const _interopRequireWildcard = require(\\"@swc/helpers/lib/_interop_require_wildcard.js\\").default;
-const _b = require(\\"../packages/b/src/index\\");
+const _b = require(\\"../packages/b/src/index.ts\\");
 async function display() {
-    const displayA = await Promise.resolve().then(()=>/*#__PURE__*/ _interopRequireWildcard(require(\\"../packages/a/src/index\\"))).then((c)=>c.displayA);
+    const displayA = await Promise.resolve().then(()=>/*#__PURE__*/ _interopRequireWildcard(require(\\"../packages/a/src/index.ts\\"))).then((c)=>c.displayA);
```
Comment (nissy-dev) [inline node-swc/__tests__/transform/issue_4730_test.mjs:42]: After applying patch dependencies correctly, this test failed. I seem #6339 causes this issue. Therefore, I updated snapshot.

**Review by kdy1 — APPROVED** [review]
swc-bump:
 - swc_ecma_ast

File: crates/swc_ecma_ast/src/expr.rs [hunk 3]
```diff
@@ -157,8 +157,8 @@ pub enum Expr {
     #[tag("TsInstantiation")]
     TsInstantiation(TsInstantiation),
 
-    #[tag("TsSatisfactionExpr")]
-    TsSatisfaction(TsSatisfactionExpr),
+    #[tag("TsSatisfiesExpression")]
+    TsSatisfies(TsSatisfiesExpr),
```
Comment (dsherret) [inline crates/swc_ecma_ast/src/expr.rs:161]: I will no longer have Benny Benassi stuck in my head while working with this. Thank you!

**Pull request merged**